### PR TITLE
ethutil/registration: use genesis fork version

### PR DESCRIPTION
**Pull request by dB2510**
Use `GENESIS_FORK_VERSION` to compute signing domain for validator registrations as per spec.
Refer:
 - https://github.com/ethereum/builder-specs/blob/100d4faf32e5dc672c963741769390ff09ab194a/specs/bellatrix/builder.md#signing
 - https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#compute_domain

category: bug
ticket: none

### CODE REVIEW

File: eth2util/registration/registration.go [hunk 1]
```diff
@@ -77,13 +81,13 @@ func getRegistrationDomain() (eth2p0.Domain, error) {
 }
 
 // GetMessageSigningRoot returns the validator registration message signing root created by the provided parameters.
-func GetMessageSigningRoot(msg *eth2v1.ValidatorRegistration) ([32]byte, error) {
+func GetMessageSigningRoot(msg *eth2v1.ValidatorRegistration, forkVersion []byte) ([32]byte, error) {
```
Comment (corverroos) [inline eth2util/registration/registration.go:84]: Use proper types here, eth2util pacakge should use eth2 types.

Comment (corverroos) [inline eth2util/registration/registration.go:84]: root cause, we implemented signing logic without proper testing. We have existing code that generates valid registrations, please write a test that ensures this code and that code generates the same sigs

**Review by corverroos — APPROVED** [review]

File: eth2util/registration/registration.go [hunk 2]
```diff
@@ -77,13 +81,13 @@ func getRegistrationDomain() (eth2p0.Domain, error) {
 }
 
 // GetMessageSigningRoot returns the validator registration message signing root created by the provided parameters.
-func GetMessageSigningRoot(msg *eth2v1.ValidatorRegistration) ([32]byte, error) {
+func GetMessageSigningRoot(msg *eth2v1.ValidatorRegistration, forkVersion eth2p0.Version) ([32]byte, error) {
```
Comment (xenowits) [inline eth2util/registration/registration.go:84]: nit:
```suggestion
func GetMessageSigningRoot(msg *eth2v1.ValidatorRegistration, genesisForkVersion eth2p0.Version) ([32]byte, error) {
```

File: eth2util/registration/registration_test.go [hunk 3]
```diff
@@ -71,8 +83,69 @@ func TestGetMessageSigningRoot(t *testing.T) {
...
+   "pubkey": "0x86966350b672bd502bfbdb37a6ea8a7392e8fb7f5ebb5c5e2055f4ee168ebfab0fef63084f28c9f62c3ba71f825e527e"
+  },
+  "signature": "0xb101da0fc08addcc5d010ee569f6bbbdca049a5cb27efad231565bff2e3af504ec2bb87b11ed22843e9c1094f1dfe51a0b2a5ad1808df18530a2f59f004032dbf6281ecf0fc3df86d032da5b9d32a3d282c05923de491381f8f28c2863a00180"
+ }`
+
+	reg := new(eth2v1.SignedValidatorRegistration)
+	err = json.Unmarshal([]byte(registrationJSON), reg)
+	require.NoError(t, err)
+
+	forkVersion := eth2p0.Version{0x00, 0x00, 0x10, 0x20}
```
Comment (xenowits) [inline eth2util/registration/registration_test.go:123]: use goerli.ForkVersion

File: eth2util/registration/registration_test.go [hunk 4]
```diff
@@ -71,8 +83,69 @@ func TestGetMessageSigningRoot(t *testing.T) {
...
+	sigRoot, err := registration.GetMessageSigningRoot(reg.Message, forkVersion)
+	require.NoError(t, err)
+
+	// Verify given signature corresponding to message signing root.
```
Comment (xenowits) [inline eth2util/registration/registration_test.go:127]:
```suggestion
	// Verify if given signature corresponds to the message signing root.
```

File: dkg/dkg.go [hunk 5]
```diff
@@ -544,9 +544,9 @@ func signAndAggValidatorRegistrations(
 	feeRecipients []string,
 	gasLimit uint64,
 	nodeIdx cluster.NodeIdx,
-	network string,
+	forkVersion []byte,
```
Comment (xenowits) [inline dkg/dkg.go:547]: can use `forkVersion eth2p0.Version`

**Review by xenowits — APPROVED** [review]
LGTM!

File: eth2util/registration/registration.go (hunk 1 above)
Comment (dB2510) [reply eth2util/registration/registration.go]: done

**Pull request merged**